rustls/src/lib.rs

```diff
@@ -320,6 +320,7 @@
     clippy::std_instead_of_core,
     clippy::use_self,
     clippy::upper_case_acronyms,
+    clippy::unnecessary_map_or,
```

Comment:
I am surprised this causes issue with Clippy stable as this looks like it was added in Rust 1.75:

- https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_map_or

Am I missing anything here?

Reply:
Raised a clippy bug for that -- it looks like the lint was only merged two weeks ago, so it cannot be in 1.75.

Reply:
I think rustversion was already added conditionally for `read_buf` option, was hoping I could use it to check that this warning does not come back. I can remove it, though.

Reply:
This is not needed. We deny warnings in stable clippy runs (and don't for nightly clippy, as the noise is not worth it). Therefore, once this lint is in a stable version, regressions of this PR will fail CI.

The list in this file is (or should be) only for clippy lints that do not warn by default (eg, `clippy::use_self` is `allow` by default).

Reply:
I have now removed the clippy updates, rebased to just make the one-line change I would like to propose